### PR TITLE
Add support for chown syscall

### PR DESCRIPTION
**Pull request by liqinggd**

### CODE REVIEW

File: services/libs/aster-std/src/syscall/chown.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+};
+use crate::log_syscall_entry;
+use crate::prelude::*;
+use crate::process::{Gid, Uid};
+use crate::util::read_cstring_from_user;
+
+use super::SyscallReturn;
+use super::{SYS_FCHOWN, SYS_FCHOWNAT};
+
+pub fn sys_fchown(fd: FileDescripter, uid: u32, gid: u32) -> Result<SyscallReturn> {
```
Comment (tatetian) [inline services/libs/aster-std/src/syscall/chown.rs:16]: > If the owner or group is specified as -1, then that ID is not
       changed.

By using `u32` as the types, the implementation falls short of handling the corner case of `-1`.

File: services/libs/aster-std/src/fs/utils/inode.rs [hunk 2]
```diff
@@ -245,6 +251,8 @@ pub trait Inode: Any + Sync + Send {
 
     fn set_mode(&self, mode: InodeMode);
 
+    fn set_owner(&self, uid: Uid, gid: Gid);
```
Comment (tatetian) [inline services/libs/aster-std/src/fs/utils/inode.rs:254]: I think this interface is ill-defined. The `chown` system call may change either of the owner and group of a file, or both. The interface only supports changing both. So I think both methods should be added, one is `set_owner`, the other is `set_group`.

File: services/libs/aster-std/src/syscall/chown.rs (hunk 1 above)
Comment (liqinggd) [reply services/libs/aster-std/src/syscall/chown.rs:16]: Thank you for this reminder; I didn't even notice that instruction.

File: services/libs/aster-std/src/fs/utils/inode.rs (hunk 2 above)
Comment (liqinggd) [reply services/libs/aster-std/src/fs/utils/inode.rs]: Yes

File: services/libs/aster-std/src/fs/utils/inode.rs [hunk 3]
```diff
@@ -245,6 +246,14 @@ pub trait Inode: Any + Sync + Send {
 
     fn set_mode(&self, mode: InodeMode);
 
+    fn owner(&self) -> Uid;
+
+    fn set_owner(&self, uid: Uid);
+
+    fn group(&self) -> Gid;
+
+    fn set_group(&self, gid: Gid);
```
Comment (tatetian) [inline services/libs/aster-std/src/fs/utils/inode.rs:255]: Should this method be fallable?

File: services/libs/aster-std/src/fs/utils/dentry.rs [hunk 4]
```diff
@@ -410,6 +411,26 @@ impl Dentry {
         self.inode.set_mode(mode)
     }
 
+    /// Get the owner of the inode
+    pub fn inode_owner(&self) -> Uid {
```
Comment (tatetian) [inline services/libs/aster-std/src/fs/utils/dentry.rs:415]: Consider removing the `inode_` prefixes

File: services/libs/aster-std/src/fs/utils/inode.rs (hunk 3 above)
Comment (liqinggd) [reply services/libs/aster-std/src/fs/utils/inode.rs]: For some filesystems (e.g., NFS, FAT32), these methods are not supported and should return an error.

File: services/libs/aster-std/src/fs/utils/dentry.rs (hunk 4 above)
Comment (liqinggd) [reply services/libs/aster-std/src/fs/utils/dentry.rs]: Done

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**